test/TypeTraitsTest.cpp

```diff
@@ -35,56 +37,115 @@ constexpr void callLambdaWithAllVariationsOfType(const auto& lambda) {
   lambda.template operator()<const decayedT&&>();
 }
 
+template <typename Tup>
+struct TestSimiliarToAnyTypeIn {
+  struct CheckSimilar {
+    template <typename T>
+    void operator()() const {
+      static_assert(SimilarToAnyTypeIn<T, Tup>);
+    }
+  };
+
+  template <typename TupType>
+  void operator()() const {
+    callLambdaWithAllVariationsOfType<TupType>(CheckSimilar{});
+  }
```

Comment:
The diff here also gets much simpler, if we have a variant of `forEachTypeInTemplate`.. and similar utils that take type identities.

Reply:
Addressed